diff --git a/dist/chunk-ojHmM-yX.d.ts b/dist/chunk-ojHmM-yX.d.ts
index 49560531974950c2bb1ee6cca2cbe3fa6d08ab1d..920d57d1533c4fafb8e2a49c4bf16466b6524382 100644
--- a/dist/chunk-ojHmM-yX.d.ts
+++ b/dist/chunk-ojHmM-yX.d.ts
@@ -1,3 +1,4 @@
+import type { ComponentAnnotations as CSFComponentAnnotations, StoryAnnotations as CSFStoryAnnotations } from "storybook/internal/csf";
 import { ComponentProps, ComponentType, JSX } from "react";
 import { AnnotatedStoryFn, ArgTypes, Args, Args as Args$1, ArgsFromMeta, ArgsStoryFn, Canvas, ComponentAnnotations, DecoratorFunction, LoaderFunction, Parameters as Parameters$1, ProjectAnnotations, Renderer, StoryAnnotations, StoryContext, StrictArgs, StrictArgs as StrictArgs$1, WebRenderer } from "storybook/internal/types";
 import { RootOptions } from "react-dom/client";
@@ -1023,7 +1024,7 @@ interface ReactPreview<T extends AddonTypes> extends Preview<ReactTypes & T> {
  * provided in meta become optional in stories, while missing required args must be provided at the
  * story level.
  */
-interface ReactMeta<T extends ReactTypes, MetaInput extends ComponentAnnotations<T>> /** @ts-expect-error ReactMeta requires two type parameters, but Meta's constraints differ */ extends Meta<T, MetaInput> {
+interface ReactMeta<T extends ReactTypes, MetaInput extends ComponentAnnotations<T>> extends Meta<T, MetaInput & CSFComponentAnnotations<T, T["args"]>> {
   /**
    * Creates a story with a custom render function that takes no args.
    *
@@ -1069,7 +1070,7 @@ interface ReactMeta<T extends ReactTypes, MetaInput extends ComponentAnnotations
    * });
    * ```
    */
-  story<TInput extends Simplify<StoryAnnotations<T, AddMocks<T['args'], MetaInput['args']>, SetOptional<T['args'], keyof T['args'] & keyof MetaInput['args']>>>>(story: TInput /** @ts-expect-error hard */): ReactStory<T, TInput>;
+  story<TInput extends Simplify<StoryAnnotations<T, AddMocks<T['args'], MetaInput['args']>, SetOptional<T['args'], keyof T['args'] & keyof MetaInput['args']>>>>(story: TInput): ReactStory<T, TInput & StoryAnnotations<T, T["args"]>>;
   /**
    * Creates a story with no additional configuration.
    *
@@ -1097,7 +1098,7 @@ interface ReactMeta<T extends ReactTypes, MetaInput extends ComponentAnnotations
  *
  * Also includes a `Component` property for portable story compatibility.
  */
-interface ReactStory<T extends ReactTypes, TInput extends StoryAnnotations<T, T['args']>> extends Story<T, TInput> {
+interface ReactStory<T extends ReactTypes, TInput extends StoryAnnotations<T, T['args']>> extends Story<T, TInput & CSFStoryAnnotations<T, T["args"]>> {
   Component: ComponentType<Partial<T['args']>>;
 }
 //#endregion
